### ISSUE TITLE
[Listbox] Accessibility of disabled elements

### ISSUE THREAD

**Issue by jbalsas** [Shopify/polaris-react-archive#9105]
## Feature request summary

Treatment of disabled elements inside `Listbox` is different than that of similar implementations of select.

- In the cases of `Select`, both native and using Polaris' component, disabled elements are sill focusable an announced as "dimmed" by VoiceOver
- For `Listbox`, disabled elements are skipped over. This might cause a bit of congnitive dissonance because the list is announced as having X elements, but then cycling through it only a subset is reachable so... where did the others go?

| Native Select | Polaris Select | Listbox | 
| --- | --- | --- |
| <video src="https://user-images.githubusercontent.com/905006/234816765-9b662cbe-a3be-4788-8d63-191a846488ec.mov" /> | <video src="https://user-images.githubusercontent.com/905006/234816786-52b77150-765d-487f-9a4b-684878aaf246.mov " /> | <video src="https://user-images.githubusercontent.com/905006/234816807-7e660636-28fd-4174-88bf-c10eb37e3c4e.mov" /> |

## Rationale

- We often use `Listbox` as a sort of superpowered `Select`, so its semantics should be similar in terms of accessibility.
- People using screen reader technologies might be confused by the component information

I checked in the [Listbox Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/) and couldn't find a reference to disabled elements, so I guess this is non-normative and up to interpretation... 🤔

**Comment by jbalsas** [comment]
cc @chloerice, @laurkim, @voiid since you all touched on accessibility bits of the Listbox component at different times... 🤗
